### PR TITLE
Add ChaosCenter developer guide docs locally

### PR DESCRIPTION
**Pull request by kwx4957**
**What this PR does / why we need it**:
- Add sidebar and docs for developer Guide 
- Update current version docs for developer Guide 
  
Currently, the [development document](https://github.com/litmuschaos/litmus/wiki/ChaosCenter-Development-Guide) has not been updated. It will not work if build development environment for the code according to the development document.

1. There is no guide for Windows OS
2. The version of mongoDB does not match with the [version](https://github.com/bitnami/charts/tree/main/bitnami/mongodb#to-1200) in helm
    - mongodb version 4 to 5 
3.  It does not run because there are no environment variables such as GRPC_PORT and REST_PORT

[Preview Link](https://litmus-docs-git-fork-kwx4957-master-cn-webteam.vercel.app/docs/developer-guide/chaoscenter-developer-guide)

image

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Checklist:**
-   [ ] Fixes #<issue number>
-   [v] Signed the commit for DCO to be passed
-   [ ] Labelled this PR & related issue with `documentation` tag

### CODE REVIEW

File: website/docs/developer-guide/chaoscenter-developer-guide.md [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+id: chaoscenter-developer-guide
+title: ChaosCenter Developer Guide
+sidebar_label: ChaosCenter Developer Guide 
+---
+
+---
+
+## **Prerequisites**
+:::note
+This document is ietented to be implemented locally. Please do not use in dev or prod environmentss
```
Comment (neelanjan00) [inline website/docs/developer-guide/chaoscenter-developer-guide.md]:
```suggestion
This document is intended to be implemented locally. Please do not use it in dev or prod environments.
```

File: website/docs/developer-guide/chaoscenter-developer-guide.md [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+- Kubernetes 1.17 or later
+- Helm3 or Kubectl
+- Node and npm
+- Docker
+- Golang
+- Local Kubernetes Cluster (via minikube, k3s or kind)
+
+## **Control Plane**
+Backend components consist of three microservices
+1. GraphQL-Server
```
Comment (neelanjan00) [inline website/docs/developer-guide/chaoscenter-developer-guide.md]:
```suggestion
1. Backend server
```

File: website/docs/developer-guide/chaoscenter-developer-guide.md [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+2. Authentication-Server
```
Comment (neelanjan00) [inline website/docs/developer-guide/chaoscenter-developer-guide.md]:
```suggestion
2. Authentication server
```

File: website/docs/developer-guide/chaoscenter-developer-guide.md [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+
+```bash
+docker exec -it m1 mongo -port 27015
+
+config={"_id":"rs0","members":[{"_id":0,"host":"m1:27015"},{"_id":1,"host":"m2:27016"},{"_id":2,"host":"m3:27017"}]}
+
+rs.initiate(config)
+
+db.getSiblingDB("admin").createUser({user:"admin",pwd:"1234",roles:[{role:"root",db:"admin"}]});
+```
```
Comment (neelanjan00) [inline website/docs/developer-guide/chaoscenter-developer-guide.md]: We should use Docusaurus tabs for individual infra types to show them. Please refactor in all the places.

**Review by neelanjan00 — CHANGES_REQUESTED** [review]

**Comment by neelanjan00** [comment]
CC: @Saranya-jena please review to validate the content.

**Comment by neelanjan00** [comment]
@kwx4957 thanks for the PR, it looks good, please make the suggested changes.

File: website/docs/developer-guide/chaoscenter-developer-guide.md [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+export LITMUS_CHAOS_OPERATOR_IMAGE=litmuschaos/chaos-operator:3.0.0
+export LITMUS_CHAOS_RUNNER_IMAGE=litmuschaos/chaos-runner:3.0.0
+export LITMUS_CHAOS_EXPORTER_IMAGE=litmuschaos/chaos-exporter:3.0.0
+export ADMIN_USERNAME=admin
+export ADMIN_PASSWORD=litmus
+export VERSION=ci
+export HUB_BRANCH_NAME=v2.0.x
+export INFRA_DEPLOYMENTS="[\"app=chaos-exporter\", \"name=chaos-operator\", \"app=event-tracker\",\"app=workflow-controller\"]"
+export INFRA_COMPATIBLE_VERSIONS='["0.2.0", "0.1.0","ci"]'
+export DEFAULT_HUB_BRANCH_NAME=master
```
Comment (Saranya-jena) [inline website/docs/developer-guide/chaoscenter-developer-guide.md:190]: We have updated the envs in the wiki, can you update them here as well?

**Review by Saranya-jena — CHANGES_REQUESTED** [review]

**Comment by kwx4957** [comment]
@neelanjan00 thanks for reviews, i will refactor and apply suggestions changes!!

File: website/docs/developer-guide/chaoscenter-developer-guide.md (hunk 5 above)
Comment (kwx4957) [reply website/docs/developer-guide/chaoscenter-developer-guide.md:190]: yes, sure. I'll update

File: website/docs/developer-guide/chaoscenter-developer-guide.md (hunk 2 above)
Comment (Saranya-jena) [reply website/docs/developer-guide/chaoscenter-developer-guide.md]: I think it's fine to have graphql server since we have 2 backend servers

Comment (kwx4957) [reply website/docs/developer-guide/chaoscenter-developer-guide.md]: Shoud we use graphql server for the name?? 
in [docs](https://docs.litmuschaos.io/docs/architecture/architecture-summary) using graphql

**Comment by neelanjan00** [comment]
@Saranya-jena just wondering if it is a good idea to keep developer docs in the same place as end-user docs? As per convention, it should be present in the GitHub repository readme itself.

**Review by neelanjan00 — APPROVED** [review]

**Review by Jonsy13 — APPROVED** [review]

**Review by hrishavjha — APPROVED** [review]

**Comment by hrishavjha** [comment]
@kwx4957, This is added in 3.9.1 versioned docs, can you also add this in all the later versions and into master as well so that it gets into every version in future?

**Comment by kwx4957** [comment]
@hrishavjha yes, i can. i will keep updated on future versions

**Review by Saranya-jena — APPROVED** [review]

**Pull request merged**